### PR TITLE
[idp-1615] Generate typescript schema

### PR DESCRIPTION
**Pull request by PrincessMadMath**
## Description of changes
Now generate Typescript schemas based on OpenAPI. 

Mostly wrapping [openapi-typescript](https://openapi-ts.pages.dev/introduction) and re-exporting schemas types in a more user friendly way.

## Additional checks
- [X] Updated the documentation of the project to reflect the changes
- [ ] Added new tests that cover the code changes

Will wait for more user feedback before adding tests to freeze outcomes.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -76,7 +80,7 @@ pnpm lint
 
 #### Linting errors
 
-If you got linting error, most of the time, they can be fixed automatically using `eslint . --fix`, if not, follow the report provided by `pnpm lint`.
+If you got linting error, most of the time, they can be fixed automatically using `pnpm run eslint-fix`, if not, follow the report provided by `pnpm lint`.
```
Comment (alexasselin008) [inline CONTRIBUTING.md:83]: pnpm eslint-fix, no need to add "run"

File: README.md [hunk 2]
```diff
@@ -18,13 +18,176 @@ Add a script in package.json to call `@workleap/create-schemas`
...
+        detail:
+          type: string
+          nullable: true
+        instance:
+          type: string
+          nullable: true
+      additionalProperties: { }
+```
+
+It will generate this schame file:
```
Comment (alexasselin008) [inline README.md:101]: Schema

File: README.md [hunk 3]
```diff
@@ -18,13 +18,176 @@ Add a script in package.json to call `@workleap/create-schemas`
...
   }
 ```
 
+It is also fowarding all the [flags of openapi-typescript](https://openapi-ts.pages.dev/cli#flags) for more customization. Note that not all flags are supported correctly.
+
+If you plan to lookup paths with route parameters we recommend using the `--path-params-as-types` flag
+
+### Example
+
+By example from this OpenAPI document:
```
Comment (alexasselin008) [inline README.md:34]: By example, c'est tres francais "par exemple"

Comment (alexasselin008) [reply README.md:34]: Given this open api document

File: packages/create-schemas/package.json [hunk 4]
```diff
@@ -16,7 +16,7 @@
     "type": "module",
     "bin": "./dist/bin.js",
     "scripts": {
-        "debug": "node dist/bin.js arg1 arg2",
+        "debug": "node dist/bin.js",
```
Comment (alexasselin008) [inline packages/create-schemas/package.json:19]: You can remove this i guess

File: packages/create-schemas/src/astHelper.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
+// Inspired from https://openapi-ts.pages.dev/node
+
+import ts, { type InterfaceDeclaration, type PropertySignature } from "typescript";
+
+// Extract schema names from AST
+// export interface components {
+//     schemas: {
```
Comment (alexasselin008) [inline packages/create-schemas/src/astHelper.ts:7]: You should use the jsdoc format instead of // comments

**Review by heqianwang — APPROVED** [review]

**Review by heqianwang — DISMISSED** [review]

**Review by heqianwang — APPROVED** [review]

**Pull request merged**